src/org/etools/j1939tools/j1939/packets/GhgPacket.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022. Equipment & Tool Institute
+ */
+
+package org.etools.j1939tools.j1939.packets;
+
+import static org.etools.j1939_84.J1939_84.NL;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.etools.j1939_84.utils.CollectionUtils;
+import org.etools.j1939tools.bus.Packet;
+import org.etools.j1939tools.j1939.model.Spn;
+import org.etools.j1939tools.j1939.model.SpnDataParser;
+
+public class GhgPacket extends GenericPacket {
+
+    private List<GhgActiveTechnologyGroup> techGroups;
+
+    public GhgPacket(Packet packet) {
+        super(packet);
+    }
+
+    // public GhgActiveTechnologyGroup getGroup(int number) {
+    // return geActiveTechnologyGroups().stream().filter(t -> {
+    // return t == number;
+    // }).findFirst().orElse(null);
+    // }
+
+    public List<GhgActiveTechnologyGroup> getActiveTechnologyGroups() {
+        if (techGroups == null) {
+            parsePacket();
+        }
+        return techGroups;
+    }
+
+    private void parsePacket() {
+        techGroups = new ArrayList<>();
+        for (Spn spn : getSpns()) {
```

Comment:
I don't understand why we're looping over the SPNs.  We should be chunking through the data in the packet to create the technology groups.